python/tests/test_conversions.py

```diff
@@ -0,0 +1,442 @@
+    """Test raising when a conversion is attempted for a type that has no registered method"""
+
+    class FooUnknownType:
+        def __repr__(self):
+            return "FooUnknownType()"
+
+    with pytest.raises(TypeError) as exc_info:
+        conversions.from_partition_value_to_py(FooUnknownType(), "foo")
+    assert (f"Cannot convert partition string to python built-in, type FooUnknownType() not supported: 'foo'") in str(
+        exc_info.value
+    )
+
+    with pytest.raises(TypeError) as exc_info:
+        conversions.to_bytes(FooUnknownType(), "foo")
+    assert ("Cannot serialize value, type FooUnknownType() not supported: 'foo'") in str(exc_info.value)
+
+    with pytest.raises(TypeError) as exc_info:
+        conversions.from_bytes(FooUnknownType(), b"foo")
+    assert ("Cannot deserialize bytes, type FooUnknownType() not supported: b'foo'") in str(exc_info.value)
+
+
+@pytest.mark.parametrize(
+    "primitive_type, value, expected_error_message",
+    [
+        (DecimalType(7, 3), Decimal("123.4567"), "Cannot serialize value, scale of value does not match type decimal(7, 3): 4"),
+        (
+            DecimalType(18, 8),
+            Decimal("123456789.123456789"),
+            "Cannot serialize value, scale of value does not match type decimal(18, 8): 9",
+        ),
+        (
+            DecimalType(36, 34),
+            Decimal("1.23456789123456789123456789123456789"),
+            "Cannot serialize value, scale of value does not match type decimal(36, 34): 35",
+        ),
+        (
+            DecimalType(7, 2),
+            Decimal("1234567.89"),
+            "Cannot serialize value, precision of value is greater than precision of type decimal(7, 2): 9",
+        ),
+        (
+            DecimalType(17, 9),
+            Decimal("123456789.123456789"),
+            "Cannot serialize value, precision of value is greater than precision of type decimal(17, 9): 18",
+        ),
+        (
+            DecimalType(35, 35),
+            Decimal("1.23456789123456789123456789123456789"),
+            "Cannot serialize value, precision of value is greater than precision of type decimal(35, 35): 36",
+        ),
+    ],
+)
+def test_raise_on_incorrect_precision_or_scale(primitive_type, value, expected_error_message):
+    with pytest.raises(ValueError) as exc_info:
+        conversions.to_bytes(primitive_type, value)
+
+    assert expected_error_message in str(exc_info.value)
+
+
+@pytest.mark.parametrize(
+    "primitive_type, value, should_raise",
+    [
+        (IntegerType(), "123.45", True),
+        (IntegerType(), "1234567.89", True),
+        (IntegerType(), "123.00", False),
+        (IntegerType(), "1234567.00", False),
+        (LongType(), "123.45", True),
+        (LongType(), "1234567.89", True),
+        (LongType(), "123.00", False),
+        (LongType(), "1234567.00", False),
+    ],
+)
+def test_from_partition_value_to_py_raise_on_incorrect_precision_or_scale(primitive_type, value, should_raise):
```

Comment:
Can you relocate the partition tests so they're grouped together?

Reply:
Done! 👍